lcls-twincat-motion/Library/POUs/Motion/FB_MotionStage.TcPOU

```diff
@@ -69,8 +72,16 @@ IF stMotionStage.bError THEN
 	stMotionStage.bEnable := FALSE;
 END_IF
 
-// Mark done if settled, then reset the FB if there is no error
-IF stMotionStage.Axis.Status.InTargetPosition AND fbDriveVirtual.bDone THEN
+// Check the limits and cancel execution if appropriate. Without this block we have infinite error spam
+bFwdHit := stMotionStage.Axis.Status.PositiveDirection AND NOT stMotionStage.bLimAllFwd;
+bBwdHit := stMotionStage.Axis.Status.NegativeDirection AND NOT stMotionStage.bLimAllBwd;
+IF bFwdHit OR bBwdHit THEN
+	stMotionStage.bExecute := FALSE;
+END_IF
+
+// Mark done if settled or if unexpected stop, then reset the FB if there is no error
+fbExecFallingEdge(CLK:=stMotionStage.bExecute);
```

Comment:
My only issue is that have an `if` statement that checks for the falling edge of `bExecute` and inside that if statement we also set `bExecute` to False. Obviously this will work, double setting the `bExecute` isn't a huge deal. The downsides are 1) That block will fire twice on a standard move (no error or limit) 2) The code is kind of confusing. I'll leave it up to you to decide if you want to break these into two separate statements somehow?

Reply:
I agree that this is confusing. I'll think a bit about if and how to revise.

Reply:
The simplest solution is to add something like "AND NOT bDoneResolved", setting bDoneResolved to TRUE at the end of this block and to FALSE somewhere earlier in the code, perhaps when we start a move

Reply:
> 1) That block will fire twice on a standard move (no error or limit)

How does this fire twice? I am looping trying to understand. You definitely don't want that to be happening.

Reply:
If the block is entered normally (the move ends successfully with no interruptions), then the block ends by setting `bExecute` from `TRUE` to `FALSE`. This triggers the falling edge, satisfying the condition to enter the code block again. After the second pass, this doesn't happen again. No bad code is run by doing this, but I agree that it's bad practice and unclean.

Reply:
Actually, a super simple fix: just add `AND NOT stMotionStage.bDone` to the conditional